src/utils/fql.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Collection,
+  Documents,
+  ExprArg,
+  Function,
+  Get,
+  Lambda,
+  Map,
+  Paginate,
+  Var,
+} from 'faunadb';
+import { q } from '../lib/faunadb/faunadb';
+
+const { Exists, If, Delete, Update, CreateFunction, CreateRole, Role } = q;
+
+// Inspiration from https://github.com/fauna-brecht/faunadb-auth-skeleton-frontend/blob/default/fauna-queries/helpers/fql.js
+
+export const DeleteIfExists = (ref: ExprArg) => If(Exists(ref), false, Delete(ref));
```

Comment:
you are actually deleting something that doesn't exist, you are calling Delete on the "false" branch

Reply:
I added this from https://github.com/fauna-brecht/faunadb-auth-skeleton-frontend/blob/default/fauna-queries/helpers/fql.js#L14-L16 but I didn't use it myself.

I understand it should rather be:

```fql
function DeleteIfExists(ref) {
  return If(Exists(ref), Delete(ref), false)
}
```